plugins/modules/zos_mount.py

```diff
@@ -0,0 +1,995 @@
+
+        if len(src_params) > 1:
+            fullcmd = "{0} PARM\\(\\'{1}\\'\\)".format(fullcmd, src_params)
+            parmtext = "{0}\n      PARM('{1}')".format(parmtext, src_params)
+
+        if len(tag_flag) > 0:
+            fullcmd = "{0} TAG\\({1},{2}\\)".format(
+                fullcmd, tag_flag, tag_ccsid)
+            parmtext = "{0}\n      TAG({1},{2})".format(
+                parmtext, tag_flag, tag_ccsid)
+
+        if allow_uids:
+            fullcmd = fullcmd + ' SETUID'
+            parmtext = parmtext + '\n      SETUID'
+        else:
+            fullcmd = fullcmd + ' NOSETUID'
+            parmtext = parmtext + '\n      NOSETUID'
+
+        if 'NOWAIT' in opts or 'nowait' in opts:
+            fullcmd = fullcmd + ' NOWAIT'
+            parmtext = parmtext + '\n      NOWAIT'
+        else:
+            fullcmd = fullcmd + ' WAIT'
+            parmtext = parmtext + '\n      WAIT'
+
+        if 'NOSECURITY' in opts or 'nosecurity' in opts:
+            fullcmd = fullcmd + ' NOSECURITY'
+            parmtext = parmtext + '\n      NOSECURITY'
+        else:
+            fullcmd = fullcmd + ' SECURITY'
+            parmtext = parmtext + '\n      SECURITY'
+
+        if len(sysname) > 1:
+            fullcmd = "{0} SYSNAME\\({1}\\)".format(fullcmd, sysname)
+            parmtext = "{0}\n      SYSNAME({1})".format(parmtext, sysname)
+
+        if len(automove) > 1:
+            fullcmd = fullcmd + ' ' + automove
+            parmtext = parmtext + '\n      ' + automove
+            if(len(automove_list) > 1):
+                fullcmd = fullcmd + '(' + automove_list + ')'
+                parmtext = parmtext + '(' + automove_list + ')'
+        parmtext = parmtext + parmtail
+
+    if gonna_unmount:     # unmount/remount
+        fullumcmd = "tsocmd UNMOUNT FILESYSTEM\\(\\'{0}\\'\\)".format(src)
+        if len(unmount_opts) < 2:
+            unmount_opts = "NORMAL"
+            fullumcmd = fullcmd + ' ' + unmount_opts
+
+    comment = ''
+
+    if gonna_unmount:
+        if currently_mounted:
+            changed = True
+            if module.check_mode is False:
+                try:
+                    (rc, stdout, stderr) = module.run_command(
+                        fullumcmd, use_unsafe_shell=True)
+                    comment += "ran unmount command\n"
+                    currently_mounted = False
+                except Exception as err:
+                    module.fail_json(msg=str(err), stderr=str(res_args))
+            else:
+                comment += "(unmount) NO Action taken: ANSIBLE CHECK MODE\n"
+                stdout = 'ANSIBLE CHECK MODE'
+        else:
+            comment += "Unmount called on data set that is not mounted.\n"
+
+    if gonna_mount:
+        if currently_mounted is False:
+            changed = True
+            if module.check_mode is False:
+                try:
+                    (rc, stdout, stderr) = module.run_command(
+                        fullcmd, use_unsafe_shell=True)
```

Comment:
Same thing as earlier comment, we have to properly quote input when using unsafe shell, not sure if this will require changes to your string building for TSO MOUNT command or not

Reply:
If quoting results in command execution issues, we should at the very least ensure we have done our due diligence and verified all of the user provided arguments follow strict formats.

Reply:
I've gone through the documentation, and don't see why I used 'unsafeShell'... I only needed that for the line counter, which I'm replacing.  I've switched them all to use_unsafe_shell=False, and will rerun tests before commit.